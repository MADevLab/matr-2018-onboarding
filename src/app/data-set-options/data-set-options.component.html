<mat-form-field>
  <mat-select [formControl]="dataSetControl"
              placeholder="Select a data set"
              (selectionChange)="changeSet.emit($event.value)">
    <mat-option *ngFor="let o of dataSetOptions"
                [value]="o.value">{{o.label}}</mat-option>
  </mat-select>
</mat-form-field>
<mat-form-field>
  <mat-select [formControl]="dateControl"
              placeholder="View data set at..."
              (selectionChange)="changeDate.emit($event.value)">
    <mat-option *ngFor="let o of dateOptions"
                [value]="o.value">{{o.label}}</mat-option>
  </mat-select>
</mat-form-field>
