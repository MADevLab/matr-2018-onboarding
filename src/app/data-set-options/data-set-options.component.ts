import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { DataSetOption } from '../data-set-option.model';
import { DataPoints, DataSet } from '../data-set.model';

@Component({
  selector: 'app-data-set-options',
  templateUrl: './data-set-options.component.html',
  styleUrls: ['./data-set-options.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DataSetOptionsComponent implements OnChanges {
  @Input() public dataSets: Map<number, DataSetOption>;

  @Output() public changeSet = new EventEmitter<DataSet>();

  @Input() public chosenSet: DataSet;

  @Output() public changeDate = new EventEmitter<number>();

  public dataSetControl = new FormControl({value: -1}, Validators.min(0));

  public dateControl = new FormControl({value: -1, disabled: true}, Validators.min(0));

  public dataSetOptions: OptionEntry[];

  public dateOptions: OptionEntry[];

  public ngOnChanges(changes: SimpleChanges) {
    this.dateOptions = [];

    if (changes.dataSets) {
      this.onDataSetsChange();
    }

    if (changes.chosenSet && this.chosenSet) {
      this.onDataSetSelected();
    }
  }

  public onDataSetsChange() {
    throw new Error('Missing Logic');
  }

  public onDataSetSelected() {
    this.dateControl.enable();
    this.dateOptions = this.chosenSet.data
      .map(this.mapDataPoint)
      .sort(this.sortDateOptions);

    // make sure the listener knows this date exists for the set chosen
    if (this.dateControl.value !== -1 && this.dateOptions.findIndex(date => date.value === this.dateControl.value) !== -1) {
      throw new Error('Missing Logic');
    } else {
      // or let them know to hold 'em
      throw new Error('Missing Logic');
    }
  }

  private mapDataPoint = (dataPoint: DataPoints) => {
    return {
      value: dataPoint.period,
      label: dataPoint.date
    };
  }

  private sortDateOptions = (a: OptionEntry, b: OptionEntry) => {
    if (a.value > b.value) {
      return 1;
    } else if (a.value < b.value) {
      return -1;
    }

    return 0;
  }
}

interface OptionEntry {
  value: number;
  label: string;
}
