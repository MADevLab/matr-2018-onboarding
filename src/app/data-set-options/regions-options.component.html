<mat-form-field>
  <mat-select multiple
              [formControl]="regionControl"
              placeholder="Filter displayed regions"
              (selectionChange)="onRegionsChange($event.value)">
    <mat-option *ngFor="let o of regions"
                [value]="o.id">{{o.title}}</mat-option>
  </mat-select>
</mat-form-field>
