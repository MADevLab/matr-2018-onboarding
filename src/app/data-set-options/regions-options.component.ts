import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';
import { FormControl } from '@angular/forms';
import { Regions } from '../regions.model';

@Component({
  selector: 'app-regions-options',
  templateUrl: './regions-options.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class RegionsOptionsComponent {
  @Input() public regions: Regions[];

  @Output() public changeRegions = new EventEmitter<Regions[]>();

  public regionControl = new FormControl();

  public onRegionsChange(ids: string[]) {
    throw new Error('Missing Logic');
  }
}
