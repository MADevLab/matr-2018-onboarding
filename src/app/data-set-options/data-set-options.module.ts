import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatListModule } from '@angular/material/list';
import { MatSelectModule } from '@angular/material/select';

import { DataSetOptionsComponent } from './data-set-options.component';
import { RegionsOptionsComponent } from './regions-options.component';

@NgModule({
  imports: [CommonModule, ReactiveFormsModule, MatListModule, MatFormFieldModule, MatSelectModule],
  declarations: [
    DataSetOptionsComponent,
    RegionsOptionsComponent
  ],
  exports: [
    DataSetOptionsComponent,
    RegionsOptionsComponent
  ]
})
export class DataSetOptionsModule {}
