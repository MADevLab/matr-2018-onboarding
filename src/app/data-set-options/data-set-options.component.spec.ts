import { SimpleChange } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatListModule } from '@angular/material/list';
import { MatSelectModule, MatSelect } from '@angular/material/select';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { DataSetOptionsComponent } from './data-set-options.component';
import { DataSetOption } from '../data-set-option.model';

describe('DataSetOptionsComponent', () => {
  let component: DataSetOptionsComponent;
  let fixture: ComponentFixture<DataSetOptionsComponent>;

  beforeEach(async () => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        ReactiveFormsModule,
        NoopAnimationsModule,
        MatFormFieldModule,
        MatListModule,
        MatSelectModule
      ],
      declarations: [ DataSetOptionsComponent ]
    });
    TestBed.compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(DataSetOptionsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('is defined', () => {
    expect(component).toBeDefined();
  });

  it('html matches the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('by default', () => {
    it('data control is disabled', () => {
      const select = fixture.debugElement.queryAll(By.directive(MatSelect))[1];

      expect(component.dateControl.disabled).toBeTruthy();
      expect(select.attributes['aria-disabled']).toBe('true');
    });
  });

  describe('on changes', () => {
    let _dataSetChangeSpy: jest.SpyInstance<any>;
    let _dataSetSelectSpy: jest.SpyInstance<any>;

    beforeEach(() => {
      _dataSetChangeSpy = jest.spyOn(component, 'onDataSetsChange');
      _dataSetSelectSpy = jest.spyOn(component, 'onDataSetSelected');
    });

    it('resets the dateOptions', () => {
      component.dateOptions = [
        {label: '2010Q1', value: 1},
        {label: '2010Q1', value: 2},
        {label: '2010Q1', value: 3},
        {label: '2010Q1', value: 4}
      ];

      component.ngOnChanges({});

      expect(component.dateOptions).toEqual([]);
    });

    it('calls onDataSetsChange, if change is to dataSets', () => {
      const change = {
        dataSets: new SimpleChange('foo', 'bar', false)
      };

      component.ngOnChanges(change);

      expect(_dataSetChangeSpy).toHaveBeenCalled();
    });

    it('calls onDataSetSelected, if change is to chosenSet', () => {
      const change = {
        chosenSet: new SimpleChange('foo', 'bar', false)
      };

      component.chosenSet = _dataSet1;
      component.ngOnChanges(change);

      expect(_dataSetSelectSpy).toHaveBeenCalled();
    });
  });

  describe('on setting data sets', () => {
    beforeEach(() => {
      component.dataSetOptions = [
        {
          value: 1,
          label: 'Foo'
        },
        {
          value: 2,
          label: 'Bar'
        }
      ];
      component.dataSetControl.setValue(-2);
      component.dateControl.setValue(-2);
      component.dateControl.enable();
    });

    it('empties out the dataSetOptions', () => {
      component.onDataSetsChange();

      expect(component.dataSetOptions).toEqual([]);
    });

    it('sets the dataSetControl value to -1', () => {
      component.onDataSetsChange();

      expect(component.dataSetControl.value).toBe(-1);
    });

    it('sets the dateControl value to -1', () => {
      component.onDataSetsChange();

      expect(component.dateControl.value).toBe(-1);
    });

    it('disables the dateControl', () => {
      component.onDataSetsChange();

      expect(component.dateControl.enabled).toBeFalsy();
    });

    it('pushes the values into dataSetOptions, if dataSets are defined', () => {
      component.dataSets = new Map<number, DataSetOption>();
      component.dataSets.set(0, _dataSetOption1);
      component.dataSets.set(1, _dataSetOption2);
      component.onDataSetsChange();

      expect(component.dataSetOptions).toMatchSnapshot();
    });
  });

  describe('on selection of a data set', () => {
    beforeEach(() => {
      component.dataSets = new Map<number, DataSetOption>();
      component.dataSets.set(0, _dataSetOption1);
      component.dataSets.set(1, _dataSetOption2);
      component.chosenSet = _dataSet1;
      component.onDataSetsChange();
    });

    it('enables the dateControl', () => {
      component.onDataSetSelected();

      expect(component.dateControl.disabled).toBeFalsy();
    });

    it('parses and sorts the dates out of the dataset', () => {
      component.onDataSetSelected();

      expect(component.dateOptions).toMatchSnapshot();
    });

    it('changeDate emits the dateControl value, if the control has a value and that value is in the data set', () => {
      const spy = jest.spyOn(component.changeDate, 'emit');

      component.dateControl.setValue(665);
      component.onDataSetSelected();

      expect(spy).toHaveBeenCalled();
      expect(spy).toMatchSnapshot();
    });

    it('changeDate emits -1 if the dateControl has no value', () => {
      const spy = jest.spyOn(component.changeDate, 'emit');

      component.onDataSetSelected();

      expect(spy).toHaveBeenCalled();
      expect(spy).toMatchSnapshot();
    });

    it('changeDate emits -1 if the dateControl value is not in the dataset', () => {
      const spy = jest.spyOn(component.changeDate, 'emit');

      component.dateControl.setValue(2);
      component.onDataSetSelected();

      expect(spy).toHaveBeenCalled();
      expect(spy).toMatchSnapshot();
    });

    it('if the dateControl value is not in the dataset, it is set to -1', () => {
      component.dateControl.setValue(2);

      const spy = jest.spyOn(component.dateControl, 'setValue');

      component.onDataSetSelected();

      expect(spy).toHaveBeenCalled();
      expect(spy).toMatchSnapshot();
    });
  });
});

const _dataSet1 = {
  concept: 'FGDP$',
  description: 'Gross State Product: Total, (Bil. Chained 2009 $, SAAR)',
  source: 'U.S. Bureau of Economic Analysis (BEA); Moody\'s Analytics Estimated and Forecasted',
  freq: 162,
  transformation: 'None',
  data: [
    {
      period: 661,
      date: '2015Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 662,
      date: '2015Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 664,
      date: '2015Q4',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 663,
      date: '2015Q3',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 665,
      date: '2016Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 666,
      date: '2016Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    }
  ]
};

const _dataSet2 = {
  concept: 'FOO',
  description: 'Another DataSet',
  source: 'U.S. Bureau of Economic Analysis (BEA); Moody\'s Analytics Estimated and Forecasted',
  freq: 162,
  transformation: 'None',
  data: [
    {
      period: 681,
      date: '2020Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 682,
      date: '2020Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 684,
      date: '2020Q4',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 683,
      date: '2020Q3',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 685,
      date: '2020Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 686,
      date: '2021Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    }
  ]
};

const _dataSetOption1 = {
  label: 'Foo',
  dataUrl: 'http://localhost/foo'
};

const _dataSetOption2 = {
  label: 'Bar',
  dataUrl: 'http://localhost/bar'
};
