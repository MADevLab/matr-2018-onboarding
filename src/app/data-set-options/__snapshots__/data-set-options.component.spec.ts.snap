// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSetOptionsComponent html matches the snapshot 1`] = `
<app-data-set-options
  changeDate={[Function EventEmitter]}
  changeSet={[Function EventEmitter]}
  dataSetControl={[Function FormControl]}
  dateControl={[Function FormControl]}
  mapDataPoint={[Function Function]}
  sortDateOptions={[Function Function]}
>
  <mat-form-field
    class="mat-form-field ng-tns-c3-0 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder ng-untouched ng-pristine ng-valid _mat-animation-noopable"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <mat-select
            aria-disabled="false"
            aria-invalid="false"
            aria-label="Select a data set"
            aria-multiselectable="false"
            aria-required="false"
            class="mat-select ng-tns-c4-1 ng-untouched ng-pristine ng-valid ng-star-inserted"
            id="mat-select-2"
            ng-reflect-form="[object Object]"
            ng-reflect-placeholder="Select a data set"
            placeholder="Select a data set"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-hidden="true"
              cdk-overlay-origin=""
              class="mat-select-trigger"
            >
              <div
                class="mat-select-value"
                ng-reflect-ng-switch="true"
              >
                
                <span
                  class="mat-select-placeholder ng-tns-c4-1 ng-star-inserted"
                >
                  Select a data set
                </span>
                
              </div>
              <div
                class="mat-select-arrow-wrapper"
              >
                <div
                  class="mat-select-arrow"
                />
              </div>
            </div>
            
          </mat-select>
          <span
            class="mat-form-field-label-wrapper"
          >
            
            <label
              aria-owns="mat-select-2"
              class="mat-form-field-label ng-tns-c3-0 mat-empty mat-form-field-empty ng-star-inserted"
              for="mat-select-2"
              id="mat-form-field-label-5"
              ng-reflect-ng-switch="false"
            >
              
              
              Select a data set
              
              
            </label>
          </span>
        </div>
        
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c3-0 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
        ng-reflect-ng-switch="hint"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c3-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field><mat-form-field
    class="mat-form-field ng-tns-c3-2 mat-primary mat-form-field-type-mat-select mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-hide-placeholder mat-form-field-disabled ng-untouched ng-pristine _mat-animation-noopable"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <mat-select
            aria-disabled="true"
            aria-invalid="false"
            aria-label="View data set at..."
            aria-multiselectable="false"
            aria-required="false"
            class="mat-select ng-tns-c4-3 ng-untouched ng-pristine mat-select-disabled ng-star-inserted"
            id="mat-select-3"
            ng-reflect-form="[object Object]"
            ng-reflect-placeholder="View data set at..."
            placeholder="View data set at..."
            role="listbox"
            tabindex="-1"
          >
            <div
              aria-hidden="true"
              cdk-overlay-origin=""
              class="mat-select-trigger"
            >
              <div
                class="mat-select-value"
                ng-reflect-ng-switch="true"
              >
                
                <span
                  class="mat-select-placeholder ng-tns-c4-3 ng-star-inserted"
                >
                  View data set at...
                </span>
                
              </div>
              <div
                class="mat-select-arrow-wrapper"
              >
                <div
                  class="mat-select-arrow"
                />
              </div>
            </div>
            
          </mat-select>
          <span
            class="mat-form-field-label-wrapper"
          >
            
            <label
              aria-owns="mat-select-3"
              class="mat-form-field-label ng-tns-c3-2 mat-empty mat-form-field-empty ng-star-inserted"
              for="mat-select-3"
              id="mat-form-field-label-7"
              ng-reflect-ng-switch="false"
            >
              
              
              View data set at...
              
              
            </label>
          </span>
        </div>
        
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c3-2 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
        ng-reflect-ng-switch="hint"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c3-2 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field>
</app-data-set-options>
`;

exports[`DataSetOptionsComponent on selection of a data set changeDate emits -1 if the dateControl has no value 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      -1,
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DataSetOptionsComponent on selection of a data set changeDate emits -1 if the dateControl value is not in the dataset 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      -1,
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DataSetOptionsComponent on selection of a data set changeDate emits the dateControl value, if the control has a value and that value is in the data set 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      665,
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DataSetOptionsComponent on selection of a data set if the dateControl value is not in the dataset, it is set to -1 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      -1,
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DataSetOptionsComponent on selection of a data set parses and sorts the dates out of the dataset 1`] = `
Array [
  Object {
    "label": "2015Q1",
    "value": 661,
  },
  Object {
    "label": "2015Q2",
    "value": 662,
  },
  Object {
    "label": "2015Q3",
    "value": 663,
  },
  Object {
    "label": "2015Q4",
    "value": 664,
  },
  Object {
    "label": "2016Q1",
    "value": 665,
  },
  Object {
    "label": "2016Q2",
    "value": 666,
  },
]
`;

exports[`DataSetOptionsComponent on setting data sets pushes the values into dataSetOptions, if dataSets are defined 1`] = `
Array [
  Object {
    "label": "Foo",
    "value": 0,
  },
  Object {
    "label": "Bar",
    "value": 1,
  },
]
`;
